src/app.js

```diff
@@ -369,7 +369,12 @@ function run () {
   // Add files received from remote instance (i.e. another browser-solidity)
   function loadFiles (filesSet) {
     for (var f in filesSet) {
-      filesProviders['browser'].set(helper.createNonClashingName(f, filesProviders['browser']), filesSet[f].content)
+      var name = helper.createNonClashingName(f, filesProviders['browser'])
+      if (filesProviders['browser'].checkSpecialChars(name)) {
```

Comment:
While this is a good counter-measure, I don't think this is a proper fix. The code using yoyo has to do proper escaping.